### DISCUSSION TITLE
Background Music Volume Control

### DISCUSSION THREAD

**Discussion by Chipperchap** [sircharlo/meeting-media-manager#203] [Ideas] [answered]
Hi, new JWMMF user here so navigating around the program.

We usually have some background music playing a few minutes out from the meeting start that like at the K-Hall is only in the background so as not to drown out conversation. (Sfl 20:23) At about 1 minute or less we mute all microphones so that the friends know the meeting is about to start but have the music continue to play. I noticed the "Kingdom Songs on Shuffle" option and had a play over ZOOM but noticed that it is (relatively) very loud. Please, how do I control the kingdom music volume without impacting on my overall volume for ZOOM?

Kind regards
Simon Winter - New Zealand

**Comment by sircharlo** [comment, marked as answer]
Very good question! Previously I had simply controlled JWMMF's volume using the system audio control panel, but there's a better way!

For the next release, I'll add a slider in the settings to change the volume of the background music, with values ranging from 5% to 100%.

**Reply by sircharlo** [reply]
Added in https://github.com/sircharlo/jw-meeting-media-fetcher/commit/82cacdd5b00c7afcf9279d09d161a97a368ab353

**Reply by sircharlo** [reply]
Will be released with the next version of JWMMF.

**Comment by Chipperchap** [comment]
Yay! I took a moment to figure out how to adjust audio in system settings but it is working great! 
Thank you for the tip. 
Thank you for considering the feature update too as it will make it easier to adjust music volume on the fly (we usually increase music volume just prior to the start of the meeting after we have engaged "Mute All" to let the friends know we are in "countdown" mode... the meeting is about to commence). 
Great program by the way. Very grateful for your immense work!

**Reply by Chipperchap** [reply]
Just letting you know that the in-app music volume control is working fine, thank you for that. At 5% it is audible but not overwhelming to compete with pre-meeting conversation.

Just a query if I may please. I am not finding the auto-stop 5 seconds out from the start of the meeting is working. I am confident that I have set it up correctly, but how is this engaged properly? 

[<img src="https://simonsheena-gmail.tinytake.com/media/1245287?filename=1647143457208_Smart+Fadeout.png&sub_type=thumbnail_preview&type=attachment&width=696&height=698" title="Powered by TinyTake Screen Capture"/><br><a href="https://www.tinytake.com">Powered by TinyTake Screen Capture</a>](url)

I thought it might relate to the settings selected meeting start time but it doesn't appear to make any difference...

[<img src="https://simonsheena-gmail.tinytake.com/media/1245288?filename=1647143577304_5+Seconds.png&sub_type=thumbnail_preview&type=attachment&width=631&height=273" title="Powered by TinyTake Screen Capture"/><br><a href="https://www.tinytake.com">Powered by TinyTake Screen Capture</a>](url)

**Reply by sircharlo** [reply]
Odd; I'll try to investigate this.

**Reply by Chipperchap** [reply]
With the latest version of JWMMF this is now fixed for me (I figured I was not using it correctly but now it allows for the playing of music many minutes out from the start of the meeting). Brilliant work and thank you so much. 

By the way, is the music streamed or are all the songbook audio files downloaded into JWMMF somewhere?

**Reply by sircharlo** [reply]
Perfect! Glad to hear it 😄 

The music is streamed for now; down the road however I'd like to change this behavior to something that would "download, save, play", to prevent re-downloading the same songs for background music at every meeting. 
Same goes for any media really; eventually we'll cache everything, and download only when a file is not already cached, or out of date.

**Reply by sircharlo** [reply]
> Perfect! Glad to hear it 😄
> 
> The music is streamed for now; down the road however I'd like to change this behavior to something that would "download, save, play", to prevent re-downloading the same songs for background music at every meeting. Same goes for any media really; eventually we'll cache everything, and download only when a file is not already cached, or out of date.

https://github.com/sircharlo/jw-meeting-media-fetcher/commit/1e0a6d5a367f5469cc26528eec7480aa719363eb implements caching for all files, to prevent needless redownloading.
